pennylane/transforms/core/transform_dispatcher.py

```diff
@@ -68,10 +70,27 @@ def __call__(self, *targs, **tkwargs):
         if callable(obj):
             return self._qfunc_transform(obj, targs, tkwargs)
 
-        raise TransformError(
-            "The object on which the transform is applied is not valid. It can only be a tape, a QNode or a qfunc."
+        # Input is not a QNode nor a quantum tape nor a device.
+        # Assume Python decorator syntax:
+        #
+        # result = some_transform(*transform_args)(qnode)(*qnode_args)
+
+        warnings.warn(
+            "The decorator syntax transform_fn(*transform_args)(qnode) has been "
+            "deprecated and will be removed in a future version. Please use either "
+            "transform_fn(qnode, *transform_args) or "
+            "functools.partial(transform_fn, *transform_args)(qnode) instead.",
+            UserWarning,
         )
 
+        if obj is not None:
+            targs = (obj, *targs)
+
+        def wrapper(obj):
+            return self(obj, *targs, **tkwargs)
+
+        return wrapper
```

Comment:
```suggestion
        partial_transform = partial(self, *targs, **tkwargs)
        partial_transform.__doc__ = f"Partial of transform {self} with bound arguemts and keyword arguments"
        return partial_transform
```

Let's give it an informative name and docstring.  And if we make it a `functools.partial`, we can inspect the bound values in a public-facing way.

Reply:
if we feel we don't _need_ to mention that it's a partial, [functools.wraps](https://docs.python.org/3/library/functools.html#functools.wraps) preserves docstrings